daemon/oci_linux.go

```diff
@@ -777,7 +777,7 @@ func WithCommonOptions(daemon *Daemon, c *container.Container) coci.SpecOpts {
 		// joining an existing namespace, only if we create a new net namespace.
 		if c.HostConfig.NetworkMode.IsPrivate() {
 			// We cannot set up ping socket support in a user namespace
-			userNS := daemon.configStore.RemappedRoot != "" && c.HostConfig.UsernsMode.IsPrivate()
+			userNS := daemon.configStore.RemappedRoot != "" || c.HostConfig.UsernsMode.IsPrivate()
```

Comment:
This seems to be effectively reverting https://github.com/moby/moby/pull/41030 ?